.github/workflows/release.yml

```diff
@@ -0,0 +1,61 @@
+name: Build and Push
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3
+        with:
+          fetch-depth: 0
+      - name: Validate HEAD Commit
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: |
+          npm i
+          npx commitlint --from HEAD~1 --to HEAD --verbose
+
+  build:
+    needs: commitlint
+    runs-on: ubuntu-22.04
+    if: ${{ contains(fromJSON('["!:", "BREAKING CHANGE:", "patch", "minor", "major"]'), github.event.head_commit.message) || !(startsWith('chore', github.event.head_commit.message) || startsWith('refactor', github.event.head_commit.message) || startsWith('docs', github.event.head_commit.message)) }}
```

Comment:
This is a complex condition, do you mind explaining in a comment what this does for future reference?

Reply:
```suggestion
    # Only run the release job if the commit message suggests a release:
    # It is assumed commits starting with "chore" | "refactor" | "docs" do not warrant a release
    if: ${{ contains(fromJSON('["!:", "BREAKING CHANGE:", "patch", "minor", "major"]'), github.event.head_commit.message) || !(startsWith('chore', github.event.head_commit.message) || startsWith('refactor', github.event.head_commit.message) || startsWith('docs', github.event.head_commit.message)) }}
```